api/osrd_infra/views/study.py

```diff
@@ -0,0 +1,204 @@
+        queryset = Study.objects.filter(project=project)
+        name = self.request.query_params.get("name", None)
+        description = self.request.query_params.get("description", None)
+        tags = self.request.query_params.get("tags", None)
+        filters = Q()
+        if name:
+            filters = filters | Q(name__icontains=name)
+        if description:
+            filters = filters | Q(description__icontains=description)
+        if tags:
+            filters = filters | Q(tags__icontains=tags)
+        return queryset.filter(filters)
+
+    def create(self, request, *args, **kwargs):
+        project = get_object_or_404(Project.objects.all(), pk=self.kwargs["project_pk"])
+        study = Study.objects.create(project_id=project.id, **request.data)
+        serializer = StudySerializer(study, context={"request": request})
+        return Response(serializer.data)
+
+    def partial_update(self, request, pk=None, project_pk=None):
+        op_study = get_object_or_404(self.get_queryset(), pk=pk)
+        op_study.project.save()  # update last_modification field of the project
+        serializer = StudySerializer(op_study, data=request.data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(data=serializer.data, status=status.HTTP_202_ACCEPTED)
+
+
+class ProjectView(
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    mixins.UpdateModelMixin,
+    generics.ListAPIView,
+    GenericViewSet,
+):
+    serializer_class = ProjectSerializer
+    filter_backends = [filters.OrderingFilter]
+    ordering_fields = ["name", "creation_date", "last_modification"]
+    search_fields = ("name", "description", "tags")
+    pagination_class = CustomPageNumberPagination
+
+    def get_queryset(self):
+        """this function will disappear to be replaced by the generic search"""
+        queryset = Project.objects.order_by("-pk")
+        name = self.request.query_params.get("name", None)
+        description = self.request.query_params.get("description", None)
+        tags = self.request.query_params.get("tags", None)
+        filters = Q()
+        if name:
+            filters = filters | Q(name__icontains=name)
+        if description:
+            filters = filters | Q(description__icontains=description)
+        if tags:
+            filters = filters | Q(tags__icontains=tags)
+        return queryset.filter(filters)
+
+    @action(url_path="image", detail=True, methods=["get"])
+    def get_image(self, request, pk=None):
+
+        queryset = Project.objects.all()
+        project = get_object_or_404(queryset, pk=pk)
+        image_db = project.image
+        if image_db:
+            stream = BytesIO(image_db)
+            image = Image.open(stream)
+            response = HttpResponse(content_type="image/png")
+            image.save(response, "PNG")
+            return response
```

Comment:
You should store images as png so you don't have to convert them later.
At the end your endpoint should look like that:

```suggestion
            return HttpResponse(image_db, content_type="image/png")
```

Reply:
And what if user upload `jpg` or `webp` ?

Reply:
It's handled at upload